File: app/src/main/java/de/metzgore/beansplan/util/di/modules/FragmentModule.java
package de.metzgore.beansplan.util.di.modules;

import dagger.Module;
import dagger.Provides;
import de.metzgore.beansplan.dailyschedule.RefreshableDailyScheduleFragment;
import de.metzgore.beansplan.weeklyschedule.WeeklyScheduleFragment;

@Module
public class FragmentModule {

    @Provides
    RefreshableDailyScheduleFragment provideRefreshableDailyScheduleFragment(RefreshableDailyScheduleFragment
                                                                                     fragment) {
        return fragment;
    }

    @Provides
    WeeklyScheduleFragment provideWeeklyScheduleFragment(WeeklyScheduleFragment fragment) {
        return fragment;
    }
}

File: app/src/main/java/de/metzgore/beansplan/util/di/modules/FragmentModule.kt
package de.metzgore.beansplan.util.di.modules

import dagger.Module
import dagger.Provides
import de.metzgore.beansplan.dailyschedule.DailyScheduleFragment
import de.metzgore.beansplan.weeklyschedule.WeeklyScheduleFragment

@Module
class FragmentModule {

    @Provides
    fun provideDailyScheduleFragment(fragment: DailyScheduleFragment): DailyScheduleFragment {
        return fragment
    }

    @Provides
    fun provideWeeklyScheduleFragment(fragment: WeeklyScheduleFragment): WeeklyScheduleFragment {
        return fragment
    }
}
